library/connect.c

```diff
@@ -50,12 +73,21 @@ static int send_fin_message(ziti_connection conn);
 
 static void process_edge_message(struct ziti_conn *conn, message *msg, int code);
 
-int ziti_channel_start_connection(struct ziti_conn *conn);
+static int ziti_channel_start_connection(struct ziti_conn *conn);
 
 static void free_handle(uv_handle_t *h) {
     free(h);
 }
 
+const char* ziti_conn_state(ziti_connection conn) {
+    return conn ? conn_state_str[conn->state] : "<NULL>";
+}
+
+static void conn_set_state(struct ziti_conn *conn, enum conn_state state) {
+    ZITI_LOG(VERBOSE, "conn[%d] transitioning %s => %s", conn->conn_id, conn_state_str[conn->state], conn_state_str[state]);
```

Comment:
why not use `ziti_conn_state` here and avoid any null pointer issue possibility?